scripts/release.sh

```diff
@@ -0,0 +1,252 @@
+        echo "Changing Version Number in $project_file"
+        # 1.4) swap new version in to file
+        sed '/MARKETING_VERSION/{s/'$current_version_number'/'$new_version_number'/;}' $project_file > $new_file
+        mv -f $new_file $project_file
+    else
+        echo "No project file change needed for version"
+    fi
+fi
+
+# 2 update version in podspec
+# SmartDeviceLink-iOS.podspec
+echo
+echo "Step 2: Update version in podspec File"
+
+# 2.1) get the current version from the podspec file
+pod_spec_file="SmartDeviceLink-iOS.podspec"
+pod_spec_new_file="NewFile.podspec"
+current_version=$(sed -n '/s.version/{s/s.version//;s/=//;s/[\"]//g;s/^[[:space:]]*//g;p;q;}' $pod_spec_file)
+echo "Current Version: "$current_version
+
+# 2.2) Use new version number from above.  If new version is different from current, change it.
+if [ $current_version != $new_version_number ]; then
+    echo "changing version in $pod_spec_file to $new_version_number"
+    # 2.3) swap new version in to file
+    sed '/s.version/{s/'$current_version'/'$new_version_number'/;}' $pod_spec_file > $pod_spec_new_file
+    mv -f $pod_spec_new_file $pod_spec_file
+fi
+
+# 3 update RPC and protocol versions
+# /SmartDeviceLink/private/SDLGlobals.m
+# extract versions and prompt user to fix it if necessary.
+echo 
+echo "Step 3: Please update RPC version in /SmartDeviceLink/private/SDLGlobals.h"
+file="SmartDeviceLink/private/SDLGlobals.m"
+current_rpc_version=$(sed -n '/SDLMaxProxyProtocolVersion/{s/^.*@//;s/[\;]//;s/[\"]//g;p;q;}' $file)
+current_protocol_version=$(sed -n '/SDLMaxProxyRPCVersion/{s/^.*@//;s/[\;]//;s/[\"]//g;p;q;}' $file)
+echo "Current RPC Version: "$current_rpc_version
+echo "Current Protocol Version: "$current_protocol_version
+echo "Step 3: If these are not correct, please update protocol versions in /SmartDeviceLink/private/SDLGlobals.m"
+read user_input
+
+
+# 4 Update to the newest BSON submodule. Update Package.swift and CocoaPods dependency files to point to latest if necessary.
+# extract version and link from Package.swift
+dependency_file="Package.swift"
+submodule_info=$(sed -n '/.package/{s/let package = Package(//;s/.package(//;s/)//;p;}' $dependency_file)
+submodule_name=$(jq -n "{$submodule_info}" | jq -r .name)
+submodule_url=$(jq -n "{$submodule_info}" | jq -r .url)
+submodule_current_version=$(jq -n "{$submodule_info}" | jq -r .from)
+
+# figure out latest version (visit link?)
+submodule_latest_version=$(gh repo view $submodule_url --json latestRelease -q .latestRelease.tagName)
+
+# compare versions
+if [ $submodule_current_version != $submodule_latest_version ]; then
+    echo
+    echo "Current version of $submodule_name: "$submodule_current_version
+    echo "Latest version of $submodule_name: "$submodule_latest_version
+    echo "Please update the submodule $submodule_name before continuing with the release."
+    echo
+    echo "You must also edit the dependancy information in SmartDeviceLink-iOS.podspec"
+    read user_input
+fi
```

Comment:
'gh' is a problem.
I like automating this part, but it would be simpler and more consistent with the rest of the script to prompt the user.  I'll draft something up.

Reply:
gh has been commented.  It we want to automate further, I think gh (GitHub cli) is the tool to use.